states/solveCache.go

```diff
@@ -0,0 +1,60 @@
+package states
+
+import (
+	"fmt"
+
+	"github.com/moby/buildkit/client/llb"
+	"github.com/pkg/errors"
+)
+
+// SolveCache is a formal version of the cache we keep mapping targets to their LLB state.
+type SolveCache map[string]llb.State
+
+// StateKeyFunc is a function that can build a key for the SolveCache. These keys seem to be
+// Highly convention based, and used elsewhere too (LocalFolders?), so this is a step at
+// formalizing that convention, since we sometimes need one key, and sometimes another.
+// It may give us some toeholds to help with some refactoring later.
+type StateKeyFunc func() string
```

Comment:
Its late; but the whole idea here was to make the key a type. An alternative here would be to "render" the data into the key, and use this as something like `type StateKey string`. I think thats better actually, how do I request a fix of my own PR? ;)